include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2993,14 +2994,45 @@ namespace internal
                     .component_dof_indices_offset[active_fe_index]
                                                  [first_selected_component];
 
+                const unsigned int n_filled_lanes =
+                  dof_info.n_vectorization_lanes_filled[dof_access_index][cell];
+
+                const bool vectorization_possible =
+                  (n_face_orientations == 1) &&
+                  (n_filled_lanes == VectorizedArrayType::size());
+
+                std::array<Number2_ *, VectorizedArrayType::size()>
+                  vector_ptrs = {};
+
+                if (vectorization_possible == false)
+                  {
+                    if (n_face_orientations == 1)
+                      {
+                        for (unsigned int v = 0; v < n_filled_lanes; ++v)
+                          vector_ptrs[v] = vector_ptr + indices[v];
+                      }
+                    else if (n_face_orientations == VectorizedArrayType::size())
+                      {
+                        for (unsigned int v = 0;
+                             v < VectorizedArrayType::size();
+                             ++v)
+                          if (cells[v] != numbers::invalid_unsigned_int)
+                            vector_ptrs[v] =
+                              vector_ptr +
+                              dof_info.dof_indices_contiguous[dof_access_index]
+                                                             [cells[v]];
+                      }
+                    else
+                      {
+                        Assert(false, ExcNotImplemented());
+                      }
+                  }
```

Comment:
Here I translate the indices to raw pointers. In the SM case, pairs of process id and index will be translated.